src/ngraph/op/parameter.hpp

```diff
@@ -45,12 +45,16 @@ namespace ngraph
                       const Shape& shape,
                       const bool cacheable = false);
 
+            void validate_and_infer_types() override;
+
             bool get_cacheable() const { return m_cacheable; }
             virtual std::shared_ptr<Node>
                 copy_with_new_args(const NodeVector& new_args) const override;
 
         protected:
             bool m_cacheable;
+            Shape m_shape;
```

Comment:
I wonder if it would be possible to set these on an output tensor descriptor. A user can always override it later?

Reply:
There isn't one until validate runs.